core/store/src/flat_state.rs

```diff
@@ -405,9 +396,41 @@ impl FlatStateDelta {
     pub fn apply_to_flat_state(self, _store_update: &mut StoreUpdate) {}
 }
 
+/// `FlatStateDelta` which uses hash of raw `TrieKey`s instead of keys themselves.
+/// Used to reduce memory used by deltas and serves read queries.
+pub struct CachedFlatStateDelta(HashMap<CryptoHash, Option<ValueRef>>);
+
+impl From<FlatStateDelta> for CachedFlatStateDelta {
+    fn from(delta: FlatStateDelta) -> Self {
+        Self(delta.0.iter().map(|(key, value)| (hash(key), value.clone())).collect())
+    }
+}
+
+impl CachedFlatStateDelta {
+    /// Assumed size in bytes for the cache entry. Estimated as 69 bytes for `CryptoHash` and `Option<ValueRef>` +
+    /// 70% hashmap overhead per entry ~= 117 bytes.
+    const ENTRY_SIZE: u64 = 117;
```

Comment:
Please avoid hardcoding such values, in my understanding rust allows compile time computations using size_of: `std::mem::size_of::< CryptoHash>() + std::mem::size_of::<Option<ValueRef>>()`